pkg/bisect/bisect.go

```diff
@@ -302,6 +310,38 @@ func (env *env) bisect() (*Result, error) {
 	return res, nil
 }
 
+func (env *env) identifyRewrittenCommit() (string, error) {
+	cfg := env.cfg
+	_, err := env.repo.CheckoutBranch(cfg.Kernel.Repo, cfg.Kernel.Branch)
+	if err != nil {
+		return cfg.Kernel.Commit, err
+	}
+	contained, err := env.repo.Contains(cfg.Kernel.Commit)
+	if err != nil || contained {
+		return cfg.Kernel.Commit, err
+	}
+
+	// We record the tested kernel commit when syzkaller triggers a crash. These commits can become
+	// unreachable after the crash was found, when the history of the tested kernel branch was
+	// rewritten. The commit might have been completely deleted from the branch or just changed in
+	// some way. Some branches like linux-next are often and heavily rewritten (aka rebased).
+	// This can also happen when changing the branch you fuzz in an existing syz-manager config.
+	// This makes sense when a downstream kernel fork rebased on top of a new upstream version and
+	// you don't want syzkaller to report all your old bugs again.
+	if cfg.Kernel.CommitTitle == "" {
+		// This can happen during a manual bisection, when only a hash is given.
+		return cfg.Kernel.Commit, fmt.Errorf(
+			"commit %v not reachable in branch '%v' and no commit title available",
+			cfg.Kernel.Commit, cfg.Kernel.Branch)
+	}
+	commit, err := env.repo.GetCommitByTitle(cfg.Kernel.CommitTitle)
```

Comment:
Is it possible that `commit == nil` and at the same time `err == nil`? If yes, it should crash a few lines below, right?

Reply:
Nope, GetCommitByTitle always gives you a valid commit or an error.

Reply:
https://github.com/HerrSpace/syzkaller/blob/4864f66d5e5ecf80c3978e4ac938def6f3c81b01/pkg/vcs/git.go#L309-L312

If I'm not mistaken, the situation described above is possible